thalerj/openbmctool.py

```diff
@@ -0,0 +1,1490 @@
+            output['OCC1'] = {'sensorName':'OCC1', 'type': 'Discrete', 'units': 'N/A', 'value': occ0, 'target': 'Active'}
+        else:
+            output['OCC0'] = {'sensorName':'OCC0', 'type': 'Discrete', 'units': 'N/A', 'value': 'Inactive', 'target': 'Inactive'}
+            output['OCC1'] = {'sensorName':'OCC1', 'type': 'Discrete', 'units': 'N/A', 'value': 'Inactive', 'target': 'Inactive'}
+        keylist = ['sensorName', 'type', 'units', 'value', 'target']
+        colWidth = setColWidth(keylist, len(colNames), output, colNames)
+        row = ""
+        outputText = ""
+        for i in range(len(colNames)):
+            if (i != 0): row = row + "| "
+            row = row + colNames[i].ljust(colWidth[i])
+        outputText += row + "\n"
+        sortedKeys = list(output.keys()).sort
+        for key in sorted(output.keys()):
+            row = ""
+            for i in range(len(output[key])):
+                if (i != 0): row = row + "| "
+                row = row + output[key][keylist[i]].ljust(colWidth[i])
+            outputText += row + "\n"
+        return outputText
+    else:
+        return res.text + occres.text
+"""
+     prints out the bmc alerts
+       
+     @param host: string, the hostname or IP address of the bmc
+     @param args: contains additional arguments used by the sel sub command
+     @param session: the active session to use
+     @param args.json: boolean, if this flag is set to true, the output will be provided in json format for programmatic consumption 
+"""     
+def sel(host, args, session):
+
+    url="https://"+host+"/xyz/openbmc_project/logging/entry/enumerate"
+    httpHeader = {'Content-Type':'application/json'}
+    #print(url)
+    res = session.get(url, headers=httpHeader, verify=False, timeout=60)
+    return res.text
+
+
+"""
+     converts a boolean value to a human readable string value
+       
+     @param value: boolean, the value to convert
+     @return: A string of "Yes" or "No"
+""" 
+def boolToString(value):
+    if(value):
+        return "Yes"
+    else:
+        return "No"
+    
+"""
+     parses the esel data and gets predetermined search terms
+       
+     @param eselRAW: string, the raw esel string from the bmc
+     @return: A dictionary containing the quick snapshot data unless args.fullEsel is listed then a full PEL log is returned
+"""    
+def parseESEL(args, eselRAW):
+    eselParts = {}
+    esel_bin = binascii.unhexlify(''.join(eselRAW.split()[16:]))
+    #search terms contains the search term as the key and the return dictionary key as it's value
+    searchTerms = { 'Signature Description':'signatureDescription', 'devdesc':'devdesc',
+                    'Callout type': 'calloutType', 'Procedure':'procedure'}
+    
+    with open('/tmp/esel.bin', 'wb') as f:
+        f.write(esel_bin)
+    errlPath = ""
+    #use the right errl file for the machine architecture
+    arch = platform.machine()
+    if(arch =='x86_64' or arch =='AMD64'):
```

Comment:
Can't we just pass an option to the tool pointing to the errl file? Also i feel like discovery of the file should be it's own function